### PR TITLE
Add Create MTO to TOO interface

### PR DESCRIPTION
**Pull request by mkrump**
## Description

This PR adds an endpoint for the creation of the MTO. It's been assumed that the MTO will already exist at this point and its "creation" actually corresponds to changing its status in such a way so that it's now viewable by the PRIME (this status is called "DRAFT" at the moment). 

## Reviewer Notes

It's still somewhat unclear to me where the various responsibilities lie and which models contain one another. I left several TODOs along with some questions, so definitely weigh in on those.

Right now the frontend just returns static data for the customers, but you can simulate generating an MTO by visiting http://officelocal:3000/too/customer/6ac40a00-e762-4f5f-b08d-3ea72a8e4b63/details and clicking "Generate MTO".

Also you can explore the endpoint at http://milmovelocal:3000/swagger-ui/ghc.html#/moveTaskOrder/updateMoveTaskOrderStatus.

## Setup

```sh
make server_test
```

## Code Review Verification Steps
* [ ] Request review from a member of a different team.
* [ ] Have the Pivotal acceptance criteria been met for this change?

## References

* [Pivotal story](https://www.pivotaltracker.com/n/projects/2181745/stories/169156353) for this change

### CODE REVIEW

File: src/shared/Entities/schema.js [hunk 1]
```diff
@@ -111,8 +111,20 @@ export const storageInTransits = new schema.Array(storageInTransit);
...
 // Entitlements
 export const entitlements = new schema.Entity('entitlements');
 
+// Move Task Orders
+export const moveTaskOrder = new schema.Entity('moveTaskOrders');
+
+moveTaskOrder.define({
+  service_item: serviceItems,
+  delivery_address: address,
+  partial_sit_delivery_address: address,
```
[the remark this answers is not in the document]
Comment (mkrump) [reply src/shared/Entities/schema.js]: honestly, not sure. I didn't know if the entitlements are going to reside on their own or be on the mto. the pivotal ticket fields didn't fully match up with the api defn of entitlements, so I wasn't sure how this all tied together.

[the remark this answers is not in the document]
Comment (mkrump) [reply src/shared/Entities/schema.js]: just looking at the rest of your comments. it seems like entitlements will be a separate table, and link into the MTO. So for the entitlements part of the customer details page, will that still be fetching the entitlements separately or will it be going through the MTO endpoint?

[the remark this answers is not in the document]
Comment (ralren) [reply src/shared/Entities/schema.js]: I personally think it should be fetched through the MTO endpoint but there may be reasons we're not aware of for having a separate entitlements endpoint. @chrisrcoles or @jim, do you two happen to know?

File: src/scenes/Office/TOO/customerDetails.jsx [hunk 2]
```diff
@@ -1,7 +1,8 @@
 import React from 'react';
 import { connect } from 'react-redux';
-import { get } from 'lodash';
-import { getEntitlements, getCustomerInfo } from 'shared/Entities/modules/moveTaskOrders';
+import { get, isEmpty } from 'lodash';
+import { getEntitlements, updateMoveTaskOrderStatus, getCustomerInfo } from 'shared/Entities/modules/moveTaskOrders';
+import { selectMoveTaskOrder } from 'shared/Entities/modules/moveTaskOrders';
```
[the remark this answers is not in the document]
Comment (mkrump) [reply src/scenes/Office/TOO/customerDetails.jsx]: 👍 f49108f84

File: src/shared/Entities/schema.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (mkrump) [reply src/shared/Entities/schema.js]: ok. based on what i heard in our various team meetings, I'm just going to nest entitlements inside mto, even though that might get moved somewhere else eventually?

File: src/scenes/Office/TOO/customerDetails.jsx [hunk 3]
```diff
@@ -58,19 +63,66 @@ class CustomerDetails extends React.Component {
...
+              <dt>Requested Pickup Date</dt>
+              <dd>{get(moveTaskOrder, 'requestedPickupDate')}</dd>
+              <dt>Customer Remarks</dt>
+              <dd>{get(moveTaskOrder, 'remarks')}</dd>
+              <dt>Service Items</dt>
+              <dd>{JSON.stringify(get(moveTaskOrder, 'serviceItems'))}</dd>
+              <dt>Status</dt>
+              <dd>{get(moveTaskOrder, 'status')}</dd>
+              <dt>Weight Entitlement</dt>
+              {/*TODO do the entitlements belong on MTO? Was thinking yes based on the ticket, but since have a separate endpoint wasn't sure*/}
```
[the remark this answers is not in the document]
Comment (mkrump) [reply src/scenes/Office/TOO/customerDetails.jsx]: added in d83335e7d, f1540eace

File: src/scenes/Office/TOO/customerDetails.jsx [hunk 4]
```diff
@@ -38,6 +42,7 @@ class CustomerDetails extends React.Component {
               <dd>{customer.destination_duty_station}</dd>
               <dt>Pickup Address</dt>
               <dd>{customer.pickup_address}</dd>
+              {/* TODO does this belong on customer or something else?*/}
```
[the remark this answers is not in the document]
Comment (mkrump) [reply src/scenes/Office/TOO/customerDetails.jsx]: ok. removed in f1540ea

File: migrations/20191023200010_create_mto_required_fields_tables.sql [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+    ADD COLUMN origin_duty_station_id uuid REFERENCES duty_stations,
+    ADD COLUMN destination_duty_station_id uuid REFERENCES duty_stations,
+    ADD COLUMN pickup_address_id uuid REFERENCES addresses,
+    ADD COLUMN destination_address_id uuid REFERENCES addresses,
+    ADD COLUMN entitlement uuid REFERENCES entitlements,
+    ADD COLUMN actual_weight integer,
+    ADD COLUMN requested_pickup_date date,
+    ADD COLUMN customer_remarks text,
+    ADD COLUMN type move_task_order_type,
+    ADD COLUMN status move_task_order_status;
```
[the remark this answers is not in the document]
Comment (mkrump) [reply migrations/20191023200010_create_mto_required_fields_tables.sql]: updated name in f1540ea

**Pull request merged**